sites/docs/src/content/docs/tutorials/tests_and_test_data/nf-test_writing_tests.md

```diff
@@ -128,24 +136,44 @@ nextflow_process {
     tag "abricate"
     tag "abricate/summary"
 
+    setup {
+        run("ABRICATE_RUN") {
+            script "../../run/main.nf"
+            process {
+            """
+            input[0] = Channel.fromList([
+                            tuple([ id:'test1', single_end:false ], // meta map
+                                file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/genome/genome.fna.gz', checkIfExists: true)),
+                            tuple([ id:'test2', single_end:false ],
+                                file(params.modules_testdata_base_path + 'genomics/prokaryotes/haemophilus_influenzae/genome/genome.fna.gz', checkIfExists: true))
+                        ])
+            """
+        }
+        }
+    }
+
     test("bacteroides_fragilis - genome_fna_gz") {
 
-        setup {
-            run("ABRICATE_RUN") {
-                script "../../run/main.nf"
-                process {
+        when {
+            process {
                 """
-                input[0] = Channel.fromList([
-                                tuple([ id:'test1', single_end:false ], // meta map
-                                    file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/genome/genome.fna.gz', checkIfExists: true)),
-                                tuple([ id:'test2', single_end:false ],
-                                    file(params.modules_testdata_base_path + 'genomics/prokaryotes/haemophilus_influenzae/genome/genome.fna.gz', checkIfExists: true))
-                            ])
+                input[0] = ABRICATE_RUN.out.report.collect{ meta, report -> report }.map{ report -> [[ id: 'test_summary'], report]}
                 """
             }
-            }
         }
 
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out).match() }
```

Comment:
I would also recommend to use the following, to easily debug versions issues between containers on the github action.
```suggestion
                { assert snapshot(
                    process.out,
                    path(process.out.versions[0]).yaml
                ).match() }
```